network-api/networkapi/wagtailpages/pagemodels/campaigns.py

```diff
@@ -436,10 +435,20 @@ class BanneredCampaignPage(PrimaryPage):
 
     show_in_menus_default = True
 
+    def get_intro(self):
+        if self.use_intro_from_parent:
+            parent = self.get_parent()
+            if parent:
+                return parent.specific.intro
```

Comment:
I believe this could raise an exception if the bannered page is nested under a page that does not have the `intro` field. What do you think about adding a test for that case? Should we return an empty string for that case?

Reply:
Sure! I've modified this now and added a test for the case where the parent intro is None

Reply:
Sorry, I should have been clearer. I meant the case where the banner page is a child of a different page type, specifically one that does not have the `intro` field (e.g. `Homepage`).

Reply:
But the test you added is good nonetheless. There should really be one test for each branch of each if statement. Which is also why I like to fail/return early if we know we can not get further.

Reply:
Thanks for explaining further, i've made some changes to this now that should hopefully be sufficient